unlock-app/src/components/interface/locks/Manage/elements/LockPicker.tsx

```diff
@@ -0,0 +1,169 @@
+
+  return (
+    <div className="flex items-center justify-center overflow-hidden bg-gray-200 rounded-full w-7 h-7">
+      <img
+        src={lockImage}
+        alt={lockAddress}
+        className="object-cover w-full h-full bg-center"
+      />
+    </div>
+  )
+}
+
+export const LockPicker = ({
+  owner,
+  onChange,
+  defaultValues,
+}: LockPickerProps) => {
+  const { network: connectedNetwork } = useAuth()
+  const [lockAddress, setLockAddress] = useState<any>(undefined)
+  const [network, setNetwork] = useState<any>(connectedNetwork!)
+  const [showCustomLockAddress, setShowCustomLockAddress] = useState(false)
+  const [customLockAddress, setCustomLockAddress] = useState('')
+
+  const getLocksByNetwork = async () => {
+    if (!network) return null
+
+    const service = new SubgraphService()
+    return (
+      (await service.locks(
+        {
+          first: 1000,
+          where: {
+            lockManagers_contains: [owner!],
+          },
+        },
+        {
+          networks: [`${network!}`],
+        }
+      )) ?? []
+    )
+  }
+
+  const { isLoading: isLoadingLocksByNetwork, data: locksByNetwork = [] } =
+    useQuery([network, owner, connectedNetwork], async () =>
+      getLocksByNetwork()
+    )
+
+  const networkHasLocks = (locksByNetwork ?? [])?.length > 0
+  const networksOptions = Object.entries(networks).map(
+    ([id, { name: label }]: [string, any]) => ({
+      label,
+      value: id,
+    })
+  )
+
+  const locksOptions: any = locksByNetwork?.map(({ address, name }: any) => {
+    const disabled = Object.keys(locksByNetwork)?.find(
+      (lockAddress: string) =>
+        lockAddress?.toLowerCase() === address?.toLowerCase()
+    )
+    return {
+      prepend: <LockImage lockAddress={address} />,
+      label: `${name}`,
+      value: address,
+      append: addressMinify(address),
+      disabled,
+    }
+  })
+
+  useEffect(() => {
+    if (typeof onChange === 'function') {
+      onChange(lockAddress, network)
+    }
+  }, [lockAddress, network, onChange])
+
+  return (
+    <div className="flex flex-col w-full gap-2">
+      <Select
+        label="Network"
+        options={networksOptions}
+        size="small"
+        defaultValue={defaultValues?.network ?? connectedNetwork}
+        onChange={setNetwork}
+      />
+      {isLoadingLocksByNetwork ? (
+        <SelectPlaceholder />
+      ) : (
+        <>
+          {showCustomLockAddress ? (
+            <Input
+              label="Custom address"
+              size="small"
+              onChange={(e) => setCustomLockAddress(e?.target?.value ?? '')}
+            />
+          ) : (
+            <>
+              {networkHasLocks ? (
+                <Select
+                  label="Lock"
+                  options={locksOptions}
+                  size="small"
+                  onChange={setLockAddress}
+                  defaultValue={defaultValues?.lockAddress}
+                />
+              ) : (
+                network && (
```

Comment:
We should always show the select, even when lock have been deployed with the option to ever another address...
Abd we should also make the link to deploy a new one visible at all times

Reply:
**Other option**
<img width="561" alt="Screenshot 2022-11-29 at 12 39 31" src="https://user-images.githubusercontent.com/20865711/204519785-1f863cd5-30b3-43a1-bac2-6ffa51b62690.png">

**Custom option**
<img width="535" alt="Screenshot 2022-11-29 at 12 39 34" src="https://user-images.githubusercontent.com/20865711/204519802-048ca543-8496-4bc3-8f6c-4fe1ec6b0a1e.png">

**Confirm custom option**
<img width="577" alt="Screenshot 2022-11-29 at 12 39 48" src="https://user-images.githubusercontent.com/20865711/204519822-435e6ea1-efd2-4f7a-afc4-d3d5a8d91a3e.png">

Reply:
Great!